src/main/java/model/FruitTransaction.java

```
@@ -0,0 +1,55 @@
+package model;
+
+public class FruitTransaction {
+
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"), SUPPLY("s"), PURCHASE("p"), RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation fromCode(String code) {
+            for (Operation op : Operation.values()) {
+                if (op.getCode().equals(code)) {
+                    return op;
+                }
+            }
+            throw new RuntimeException(code);
```

It's better to throw a custom exception with a more informative message instead of a generic `RuntimeException`. This can help in debugging and provide more context about what went wrong. For example: `throw new IllegalArgumentException("Unsupported operation code: " + code);`